src/window/floatingwindow.cpp

```diff
@@ -19,6 +19,41 @@ void ProxyFloatingWindow::connectWindow() {
 	this->window->setMaximumSize(this->bMaximumSize);
 }
 
+void ProxyFloatingWindow::setVisible(bool visible) {
+	if (!visible) {
+		QObject::disconnect(this->mParentVisibleConn);
+		this->ProxyWindowBase::setVisible(false);
+		return;
+	}
+
+	// If there's a parent, set transient before showing
+	if (this->mParentProxyWindow) {
+		auto* pw = this->mParentProxyWindow->backingWindow();
+		if (pw && pw->isVisible()) {
+			if (this->window) this->window->setTransientParent(pw);
+		} else {
+			// Parent not visible yet - wait for it.
+			QObject::disconnect(this->mParentVisibleConn);
+			this->mParentVisibleConn = QObject::connect(
+			    this->mParentProxyWindow,
+			    &ProxyWindowBase::backerVisibilityChanged,
+			    this,
+			    [this]() {
+				    auto* pw =
+				        this->mParentProxyWindow ? this->mParentProxyWindow->backingWindow() : nullptr;
+				    if (!pw || !pw->isVisible() || !this->window) return;
+				    this->window->setTransientParent(pw);
+				    QObject::disconnect(this->mParentVisibleConn);
+				    this->ProxyWindowBase::setVisible(true);
+			    }
+			);
```

Comment:
disconnect via filters is preferred where possible (move the lambda in setVisible to its own slot method, then disconnect with `QObject::disconnect(this->mParentProxyWindow, backingwindowvisible, ... )`.